Compatibility with MODX 3

alroniks opened the pull request:
### Что оно делает?

Реализует совместимость pdoTools с MODX 3.0.

### Зачем это нужно?

Сделать возможным использование самой популярной библиотеки с новой версией MODX.

### Связанные проблема(ы)/PR(ы)
Нет

sergant210 committed:
v.3.0.0-beta for MODX3

Ibochkarev reviewed (commented):
```shell
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/core/src/Revolution/Transport/modPackageBuilder.php : 147) Создан новый пакет с подписью: pdotools-3.0.0-beta
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/core/src/Revolution/Transport/modPackageBuilder.php : 186) Зарегистрировано пространство имён пакета: pdotools
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/core/src/Revolution/Transport/modPackageBuilder.php : 204) Пространство имён «pdotools» упаковано в пакет.
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/pdoTools/_build/build.transport.php : 50) Created Transport Package and Namespace.
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/pdoTools/_build/build.transport.php : 77) Packaged in 10 snippets.
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/pdoTools/_build/build.transport.php : 98) Packaged in 1 plugins.
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/pdoTools/_build/build.transport.php : 139) Packaged in 8 System Settings.
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/pdoTools/_build/build.transport.php : 158) Packaged in 1 Plugins events.
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/pdoTools/_build/build.transport.php : 172) Added package attributes and setup options.
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/pdoTools/_build/build.transport.php : 175) Packing up transport package zip...
[2021-12-07 03:59:22] (INFO @ /Users/bochkarev/Desktop/modx-3.0.0-beta2/pdoTools/_build/build.transport.php : 227) 

Execution time: 0.1831 s
```

После установки создается пакет в `core/packages/pdotools-3.0.0-beta.transport.zip`

Но пакет не устанавливается

![изображение](https://user-images.githubusercontent.com/2138260/144964068-cf73ef21-8b6f-4132-a79b-000bd0e333a2.png)

- Переходим в Менеджер пакетов - Искать пакеты локально
- Видим в списке пакет для установки
- Нажимаем установить

![изображение](https://user-images.githubusercontent.com/2138260/144964321-5f86965e-5cb4-401a-9a1c-0cec5c532cd4.png)

Админка падает

Уточнение не был запущен composer install

sergant210 committed:
Some refactoring for 3.0.0-beta

Commit:
Rename build assets package name to follow rules and add package lock file

Commit:
Use new way of registering constant and small impromenets in file paths

Commit:
Add MODX as dev-dependency for local development

sergant210 committed:
Update readme file.

Commit:
Merge pull request #320 from modx-pro/feature/improvements

Несколько улучшений по коду рефакторинга

Commit:
Merge pull request #321 from modx-pro/require-local-modx

Add MODX as dev-dependency for local development

ilyautkin committed:
Fix makeUrl method properties

Boshnik committed:
Update transport.plugins.php

Qowyn committed:
Fix prepareTVs and processTVs = 1 not working

prepateTVs = 1 and processTVs = 1 stopped working in v.3.0.0-Beta as the equality check changed to identity check. But all values arriving from usual snippet calls and property sets arrive as string instead of number, making the "=== 1" check always fail, except maybe if calling the snippets by using the MODX runSnippet API.

halftrainedharry committed:
Update connector.php

Commit:
Fixed bugs with getting config.

sergant210 committed:
Merge pull request #335 from Qowyn/patch-1

Fix prepareTVs and processTVs = 1 not working

Commit:
Merge branch '3x-new' of https://github.com/modx-pro/pdoTools into 3x-new

Commit:
Merge branch '3x-new' of https://github.com/modx-pro/pdoTools into 3x-new

sergant210 committed:
Merge pull request #336 from halftrainedharry/patch-1

[3.0.0-beta] fatal error in connector.php

Commit:
Merge branch '3x-new' of https://github.com/modx-pro/pdoTools into 3x-new

sergant210 committed:
Merge pull request #322 from modx-pro/fix-makeurl

Fix makeUrl method properties

Commit:
Merge branch '3x-new' of https://github.com/modx-pro/pdoTools into 3x-new

sergant210 committed:
Merge pull request #326 from Boshnik/patch-1

Замена имени класса на FQN.

Commit:
Fixed fatal error caused by loop (#341).

Commit:
Fixed checking TV before adding to the query (#343).

Commit:
Changed $_SERVER['DOCUMENT_ROOT'] to dirname in the connector (#332).

Commit:
v.3.0.0-pl

Commit:
Fixed bug when clearing cache (#347).

Commit:
Fixed incorrect next and prev canonical URLs (#318).

Commit:
Added aliases "pdoTools" and "pdoFetch" for the service container.

Commit:
v.3.0.1-pl

Commit:
Fixed a bug in the Fenom parser (#331).

bezumkin committed:
Поддержка файловых сниппетов в pdoPage

[С некоторых пор](https://github.com/modx-pro/pdoTools/commit/ab4934d190c67622dcc4b005193249da0ada07f0#diff-df1b56538c6abe3aefac65ce5f4a027cc988cb15e21882bb272cdcb06281c2f5R142) `pdoPage` вызывает все сниппеты в `element` некэшируемыми, принудительно подставляя восклицательный знак перед именем.

Это ломает работу файловых сниппетов:
```
{'!pdoPage' | snippet : [
    'element' => '@FILE elements/snippets/overview.snippet.php',
    'tplPage' => '@FILE elements/chunks/pagination/page.tpl',
    'tplPageWrapper' => '@FILE elements/chunks/pagination/page-wrapper.tpl'
]}
```

Имя получается `!@FILE elements...`, собачка на 1й позиции считается за Property Set и сниппет не может быть загружен.

Данное изменение это исправляет, позволяя указывать восклицательный знак и для элементов c биндингами.

sergant210 committed:
Merge pull request #360 from modx-pro/uncached-bindings

Поддержка файловых сниппетов в pdoPage

Commit:
Merge remote-tracking branch 'origin/3x-new' into 3x-new

Commit:
Security fix for the pdoPage snippet.

Commit:
v.3.0.2-pl.
